testtools/iothub_test/src/iothub_account.c

```diff
@@ -0,0 +1,1149 @@
+{
+    HTTP_HEADERS_HANDLE httpHeader = HTTPHeaders_Alloc();
+    if (httpHeader != NULL)
+    {
+        if (HTTPHeaders_AddHeaderNameValuePair(httpHeader, "Authorization", " ") != HTTP_HEADERS_OK ||
+            HTTPHeaders_AddHeaderNameValuePair(httpHeader, "User-Agent", "Microsoft.Azure.Devices/1.0.0") != HTTP_HEADERS_OK ||
+            HTTPHeaders_AddHeaderNameValuePair(httpHeader, "Accept", "application/json") != HTTP_HEADERS_OK ||
+            HTTPHeaders_AddHeaderNameValuePair(httpHeader, "Content-Type", "application/json; charset=utf-8") != HTTP_HEADERS_OK)
+        {
+            LogError("Failure adding http headers.\r\n");
+            HTTPHeaders_Free(httpHeader);
+            httpHeader = NULL;
+        }
+        else
+        {
+            if (appendIfMatch)
+            {
+                if (HTTPHeaders_AddHeaderNameValuePair(httpHeader, "If-Match", "*") != HTTP_HEADERS_OK)
+                {
+                    LogError("Failure adding if-Match http headers.\r\n");
+                    HTTPHeaders_Free(httpHeader);
+                    httpHeader = NULL;
+                }
+            }
+        }
+    }
+    return httpHeader;
+}
+
+static int generateDeviceName(const char* prefix, char** deviceName)
+{
+    int result;
+    char deviceGuid[DEVICE_GUID_SIZE];
+    if (UniqueId_Generate(deviceGuid, DEVICE_GUID_SIZE) != UNIQUEID_OK)
+    {
+        LogError("Unable to generate unique Id.\r\n");
+        result = MU_FAILURE;
+    }
+    else
+    {
+        size_t len = strlen(prefix) + DEVICE_GUID_SIZE;
+        *deviceName = (char*)malloc(len + 1);
+        if (*deviceName == NULL)
+        {
+            LogError("Failure allocating device ID.\r\n");
+            result = MU_FAILURE;
+        }
+        else
+        {
+            if (sprintf_s(*deviceName, len + 1, prefix, deviceGuid) <= 0)
+            {
+                LogError("Failure constructing device ID.\r\n");
+                free(*deviceName);
+                result = MU_FAILURE;
+            }
+            else
+            {
+                LogInfo("Created Device %s.", *deviceName);
+                result = 0;
+            }
+        }
+    }
+    return result;
+}
+
+static int retrieveConnStringInfo(IOTHUB_ACCOUNT_INFO* accountInfo)
+{
+    int result;
+    int beginName, endName, beginIothub, endIothub, beginHost, endHost, beginKey;
+    size_t totalLen = strlen(accountInfo->connString);
+
+    if (sscanf(accountInfo->connString, "HostName=%n%*[^.]%n.%n%*[^;];%nSharedAccessKeyName=%n%*[^;];%nSharedAccessKey=%n", &beginHost, &endHost, &beginIothub, &endIothub, &beginName, &endName, &beginKey) != 0)
+    {
+        LogError("Failure determining the string length parameters.\r\n");
+        result = MU_FAILURE;
+    }
+    else
+    {
+        if ((accountInfo->iothubName = (char*)malloc(endHost - beginHost + 1)) == NULL)
+        {
+            LogError("Failure allocating iothubName.\r\n");
+            result = MU_FAILURE;
```

Comment:
Probably better to initialize result at line #149 to MU_FAILURE

Reply:
There is a reason I don't, we can talk about it afterwords.